spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java

```diff
@@ -195,20 +199,8 @@ int[] initEqDeleteRowIdMapping(int numRowsToRead) {
      */
     void applyEqDelete() {
       Iterator<InternalRow> it = columnarBatch.rowIterator();
-      int rowId = 0;
-      int currentRowId = 0;
-      while (it.hasNext()) {
-        InternalRow row = it.next();
-        if (deletes.eqDeletedRowFilter().test(row)) {
-          // the row is NOT deleted
-          // skip deleted rows by pointing to the next undeleted row Id
-          rowIdMapping[currentRowId] = rowIdMapping[rowId];
-          currentRowId++;
-        }
-
-        rowId++;
-      }
-
+      LOG.debug("Applying equality deletes to row id mapping");
+      int currentRowId = deletes.applyEqDeletes(it, rowIdMapping);
```

Comment:
Is this just a refactoring?

Reply:
Yes and no. There doesn't seem to be a natural way to get the `DeleteCounter` to `ColumnarBatchReader`. In this method, the `rowIdMapping` int array is updated by applying the equality deletes in the `DeleteFilter` `deletes`. As each row is tested, if it is deleted, we need to increment the `DeleteCounter`. The `DeleteFilter` has the `DeleteCounter`, so I moved the logic completely over to `DeleteFilter`. The alternative is to expose `DeleteFilter`'s `DeleteCounter` so that it can be used here, but I think it is better to encapsulate that in `DeleteFilter`.

Reply:
I'd prefer not to move this logic to `DeleteFilter` since it is more relevant here. We may expose DeleteFilter's DeleteCounter as you suggested.